Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,426 @@
+                    //Heat and mols account for each other, a lot of hot mols are more damaging then a few
+                    //Mols start to have a positive effect on damage after 350
+                    var MoleClamp = Math.Clamp(mixture.TotalMoles / 200f, 0.5f, 1f);
+                    var HeatDamage = ((Atmospherics.T0C + SupermatterComponent.HeatPenaltyThreshold)*SMcomponent.DynamicHeatResistance);
+                    damage = Math.Max(damage + (Math.Max(MoleClamp * mixture.Temperature - HeatDamage, 0) * SMcomponent.MoleHeatPenalty / 150) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //Power only starts affecting damage when it is above 5000
+                    damage = Math.Max(damage + (Math.Max(SMcomponent.Power - SupermatterComponent.PowerPenaltyThreshold, 0f)/500f) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //Molar count only starts affecting damage when it is above 1800
+                    damage = Math.Max(damage + (Math.Max(mixture.TotalMoles - SupermatterComponent.MolePenaltyThreshold, 0f)/80f) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //There might be a way to integrate healing and hurting via heat
+			        //healing damage
+                    if (mixture.TotalMoles < SupermatterComponent.MolePenaltyThreshold)
+                    {
+                        //Only has a net positive effect when the temp is below 313.15, heals up to 2 damage. Psycologists increase this temp min by up to 45
+                        var heatcap = ((Atmospherics.T0C + SupermatterComponent.HeatPenaltyThreshold) + (45 * SMcomponent.PsyCoeff));
+                        damage = Math.Max(-(2 * (1 - (mixture.Temperature / heatcap))),-2);
+                    }
+
+                    //if there are space tiles next to SM
+                    //TODO: change moles out for checking if adjacent tiles exist
+                    foreach(var adjacent in _atmosphere.GetAdjacentTileMixtures(Xform.Coordinates))
+                    {
+                        if (adjacent.TotalMoles == 0)
+                        {
+                            float factor;
+
+                            factor = integrity switch
+                            {
+                                (<= 25) => 0.002f,
+                                (<= 55) and (> 25) => 0.005f,
+                                (<= 75) and (> 55) => 0.0009f,
+                                (<= 90) and (> 75) => 0.0005f,
+                                _ => 0,
+                            };
+
+                            damage = Math.Clamp((SMcomponent.Power * factor) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                        }
+                    }
+                }
```

Comment:
This is unnecessary as having an adiecent space tile means there is no atmosphere on the tile the sm is on anyway.

Reply:
not necessary if monstermos isnt enabled. but i dont see why this needs to exist at all regardless of that